components/Cards/src/MDCCard.h

```diff
@@ -0,0 +1,119 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+#import "MaterialInk.h"
+#import "MaterialShadowLayer.h"
+
+@interface MDCCard : UIControl
+
+/**
+ The corner radius for the card
+ Default is set to 4.
+ */
+@property(nonatomic, assign) CGFloat cornerRadius UI_APPEARANCE_SELECTOR;
+
+/**
+ The inkView for the card that is initiated on tap
+ */
+@property(nonatomic, readonly, strong, nonnull) MDCInkView *inkView;
+
+
+/**
+ Sets the shadow elevation for an UIControlState state
+
+ @param shadowElevation The shadow elevation
+ @param state UIControlState the card state
+ */
+- (void)setShadowElevation:(MDCShadowElevation)shadowElevation forState:(UIControlState)state
+    UI_APPEARANCE_SELECTOR;
+
+/**
+ Returns the shadow elevation for an UIControlState state
+
+ If no elevation has been set for a state, the value for UIControlStateNormal will be returned.
+ Default value for UIControlStateNormal is 1
+ Default value for UIControlStateHighlighted is 8
+
+ @param state UIControlState the card state
+ @return The shadow elevation for the requested state.
+ */
+- (MDCShadowElevation)shadowElevationForState:(UIControlState)state;
+
+/**
+ Sets the border width for an UIControlState state
+
+ @param borderWidth The border width
+ @param state UIControlState the card state
+ */
+- (void)setBorderWidth:(CGFloat)borderWidth forState:(UIControlState)state
+    UI_APPEARANCE_SELECTOR;
+
+/**
+ Returns the border width for an UIControlState state
+
+ If no border width has been set for a state, the value for UIControlStateNormal will be returned.
+ Default value for UIControlStateNormal is 0
+ Default value for UIControlStateHighlighted is 0
+
+ @param state UIControlState the card state
+ @return The border width for the requested state.
+ */
+- (CGFloat)borderWidthForState:(UIControlState)state;
+
+/**
+ Sets the border color for an UIControlState state
+
+ @param borderColor The border color
+ @param state UIControlState the card state
+ */
+- (void)setBorderColor:(nullable UIColor *)borderColor forState:(UIControlState)state
+    UI_APPEARANCE_SELECTOR;
+
+/**
+ Returns the border color for an UIControlState state
+
+ If no border color has been set for a state, the value for UIControlStateNormal will be returned.
```

Comment:
Please consider using `nil` as "no borderColor" rather than an explicit `clearColor`. This matches how many UIKit color properties handle `nil` colors.  It would also match how `shadowColor` will be used within this class.  You can use `clearColor` internally if required for the implementation.

Reply:
I'm not sure if that is still the case when dealing with CALayer.  For instance, the default value of a UIButon.layer.borderColor is an instance of UIColor (not nil).  We don't typically see it because strokeWidth is set to 0.

I would consider setting the strokeColor to BaseLineColorScheme.primaryColor for .normal.  Do not specify the other states so they will fallback to the .normal behavior.